src/SprykerEco/Service/AkeneoPim/Dependencies/External/Api/Adapter/HttpClient/Promise.php

```diff
@@ -0,0 +1,178 @@
+
+            throw $this->exception;
+        });
+    }
+
+    /**
+     * @param callable|null $onFulfilled
+     * @param callable|null $onRejected
+     *
+     * @return \Http\Promise\Promise
+     */
+    public function then(?callable $onFulfilled = null, ?callable $onRejected = null): HttpPromise
+    {
+        return new static($this->promise->then($onFulfilled, $onRejected), $this->request);
+    }
+
+    /**
+     * @return string
+     */
+    public function getState(): string
+    {
+        return $this->state;
+    }
+
+    /**
+     * @param bool $unwrap
+     *
+     * @throws \Http\Client\Exception
+     *
+     * @return \Psr\Http\Message\ResponseInterface|null
+     */
+    public function wait($unwrap = true): ?ResponseInterface
+    {
+        $this->promise->wait(false);
+
+        if ($unwrap) {
+            if ($this->getState() == static::REJECTED && $this->exception !== null) {
+                throw $this->exception;
+            }
+
+            return $this->response;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param \Psr\Http\Message\ResponseInterface $response
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    protected function handleResponse(ResponseInterface $response): ResponseInterface
+    {
+        $this->response = $response;
+        $this->state = static::FULFILLED;
+
+        return $response;
+    }
+
+    /**
+     * @param callable $rejectCallback
+     * @param \Psr\Http\Message\RequestInterface $request
+     *
+     * @return \Http\Client\Exception
+     */
+    protected function handleRejectCallback(callable $rejectCallback, RequestInterface $request): HttpExceptionInterface
+    {
+        if ($rejectCallback instanceof HttpExceptionInterface) {
+            return $rejectCallback;
+        }
+
+        if ($rejectCallback instanceof GuzzleException) {
+            return $this->handleGuzzleException($rejectCallback, $request);
+        }
+
+        if ($rejectCallback instanceof Throwable) {
+            return new HttpTransferException('Invalid exception returned from Guzzle', 0, $rejectCallback);
+        }
+
+        return new UnexpectedValueException('Reason returned from Guzzle must be an Exception');
+    }
+
+    /**
+     * @param \GuzzleHttp\Exception\GuzzleException $exception
+     * @param \Psr\Http\Message\RequestInterface $request
+     *
+     * @return \Http\Client\Exception
+     */
+    protected function handleGuzzleException(GuzzleException $exception, RequestInterface $request): HttpExceptionInterface
+    {
+        if ($exception instanceof SeekException) {
+            return new HttpRequestException($exception->getMessage(), $request, $exception);
+        }
+
+        if ($exception instanceof ConnectException) {
+            return new HttpNetworkException($exception->getMessage(), $exception->getRequest(), $exception);
+        }
+
+        if ($exception instanceof RequestException) {
```

Comment:
* Let's get rid of the unnecessary level of nesting:
```suggestion
        if ($exception instanceof RequestException && $exception->hasResponse()) {
        return new HttpException(
            $exception->getMessage(),
            $exception->getRequest(),
            $exception->getResponse(),
            $exception
        );
    }

    if ($exception instanceof RequestException) {
        return new HttpRequestException($exception->getMessage(), $exception->getRequest(), $exception);
    }

```